example/generic_social_network/app/resources/follow_resource.py

```diff
@@ -0,0 +1,45 @@
+from cartographer.field_types import StringAttribute, SchemaRelationship
+from cartographer.parsers.schema_parser import SchemaParser
+from cartographer.resources.api_resource import APIResource
+from cartographer.schemas.schema import Schema
+from cartographer.serializers import SchemaSerializer
+from generic_social_network.app.models.tables.follow import Follow
+
+
+class FollowSchema(Schema):
+    SCHEMA = {
+        'type': 'follow',
+        'id': StringAttribute(resource_method='follow_id'),
+        'relationships': {
+            'follower': SchemaRelationship(model_type='user', id_attribute='follower_id'),
+            'followed': SchemaRelationship(model_type='user', id_attribute='followed_id')
+        }
+    }
+
+
+class FollowSerializer(SchemaSerializer):
+    @classmethod
+    def schema(cls):
+        return FollowSchema
+
+    def follow_id(self):
+        return "{follower_id}-{followed_id}".format(
+            follower_id=self.model.follower_id,
+            followed_id=self.model.followed_id
+        )
+
+
+class FollowParser(SchemaParser):
+    @classmethod
+    def schema(cls):
+        return FollowSchema
+
+
+class FollowResource(APIResource):
+    SCHEMA = FollowSchema
+    SERIALIZER = FollowSerializer
+    PARSER = FollowParser
+    # MASK = FollowMask
+    MODEL = Follow
+
+FollowResource.register()
```

Comment:
I was thinking something like this:

```import types

def parser_factory(parser_name, schema_class):
    new_parser = types.new_class(parser_name, bases=(SchemaParser,))
    new_parser.schema = classmethod(lambda self: schema_class)
    return new_parser

ConversationParser = parser_factory('ConversationParser', ConversationSchema)
ConversationUserParser = parser_factory('ConversationUserParser', ConversationUserSchema)```

Getting the info to talk about it in more detail required me to figure out how it's done. :D Python 3.4+ makes this super easy for us. I quickly tested it and it works - not sure how useful it will be in Patreon because our classes are often richer than just a `schema()`, but this factory would certainly help in your sample project, or perhaps as a way of reducing redundancy for other classes throughout Cartographer!